### PR TITLE
New Vault- The Syndicate Base

### PR DESCRIPTION
**Pull request by icantthinkofanameritenow**
Following the earlier feedback regarding my first attempt at making a vault, I overhauled it into a full-fledged base, crawling with guards,and turrets. Assuming you survive the way there, you might even be able to get in touch with the Syndicate yourself!

Just be sure to bring a screwdriver and multitool, several of the doors are bolted shut and can't be bypassed. Oh, and if you have to go to the bathroom make sure you don't interrupt anyone...

untitled
armory

Edits: Added a few trash items, paintings, and posters here and there for flavor, removed combat drones and the turrets in the meeting room. Top row of racks removed, replaced with closets on left side of room.

### CODE REVIEW

**Comment by NigglyWiggly** [comment]
> Wishgranter

Jesus Christ 

> Final locker has a balloon and a toy in it

D E V I L I S H

**Comment by icantthinkofanameritenow** [comment]
@NigglyWiggly
It's a Syndicate beacon like the one from Xenoarchaeology. Common mistake, they have the same sprite.

**Comment by NigglyWiggly** [comment]
Oh, neat.
Also those racks are pretty poorly placed, you can't access the second row of them without disassembling the first row

**Comment by icantthinkofanameritenow** [comment]
@NigglyWiggly
Noted. I'll go and rearrange them.

**Comment by Imagebard** [comment]
Hey, you know that rack with the megabuster?
Can't you just include the whole Megaman collection? They're admin-spawn only as of now, and its kinda boring.

**Comment by ghost** [comment]
Have you made sure the drone doesn't fight the syndicates?

**Comment by icantthinkofanameritenow** [comment]
@DrCelt
I did at first, but after being told that their shots would still destroy rwalls I decided to just remove the drones entirely. They're no longer present on the map as of my most recent revision.

**Comment by Shadowmech88** [comment]
@icantthinkofanameritenow 
You can always make a child of drones that fire non-wall-destroying projectiles, and have those spawn in the vault instead.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88 
Could work, but on the other hand the vault is already loaded with Syndicate agents. 

Speaking of which, I noticed in the code that they don't attack nuke ops (but will still attack other antags as normal). You think that might be an issue down the line?

On the one hand it means that this vault equates to a bunch of free loot for the nuke ops, but on the other hand they still have to find it first and hack the doors open, and even then there's a good chance much of the loot will be weaker or equal to the stuff they already have available to them. That, and they'd be wasting time that might be better used for actually securing the disk.

**Comment by Kitfox88** [comment]
Much better, digging it. :+1:

**Comment by Imagebard** [comment]
If the drones were removed, I wanted to ask: Were the turrets removed too?
If not, I'm curious as to what is inside of them. Stun Rounds? Normal Lasers? Laser Cannon?

**Comment by icantthinkofanameritenow** [comment]
@Imagebard
As I mentioned, the ones in the room with the beacon are still where they are. Only the ones in the room with the commandos are gone, since someone pointed out how out-of-place they looked. 

As for what's inside of them, I just used the default versions so I'd assume lasers. Is there a way to set what they're armed with? Because if so, I have some interesting ideas as to what to arm them with.

**Comment by Imagebard** [comment]
>Is there a way to set what they're armed with?

Well, telecomms turrets are armed with what I believe are laser cannons or something differing form the default ones in the AI core, considering the bolts they shot when set to lethal are different form default, so yes. You probably can.

**Comment by icantthinkofanameritenow** [comment]
@Imagebard 
I'll have to take a look at that, then. 

It'll probably mean I have to switch the turret types, it turns out I was using the one that has a pre-set projectile type. On that note, how dangerous would you estimate those turrets would be if I set them to fire energy crossbow bolts?

**Comment by Imagebard** [comment]
That would leave you wide open to being attacked by the mobs, and give you some toxins damage.
Unfortunately, those turrets are shit since mobs can't co-ordinate attack with turrets and will rush out of the room. Also, it would make the vault with the fake wish granter unprotected.

**Comment by Boxe** [comment]
they aren't going to magically fuck off from their base and fly off onto the station like spiders used to, right?

**Comment by Imagebard** [comment]
Oh, Idea.
If you get knocked down, stun turrets go down, but lethal turrets keep shotting.
If you put a lethal and a stun turret in the same room, it should work out better than using mobs.

**Comment by Shadowmech88** [comment]
It should be completely possible to make a turret that fires whatever projectile you want it to.

**Comment by icantthinkofanameritenow** [comment]
@Boxe
The inner door is bolted, they couldn't get out if they tried.

**Comment by icantthinkofanameritenow** [comment]
@Imagebard 

As it turns out, the only way to make turrets act that way also causes them not to shoot at you if you're Security or a Head. (or have the ID of one, anyway). Instead I'm going for plan B: making the front turrets stun you with ebow bolts and the back turrets blast you with pulse rifle shots. Given the obvious difficulty this would cause, I thought ahead and planted a grenade launcher and some EMP grenades in the armory room. Odds are you'll need it even with an energy shield looted from one of the commandos (or the ablative armor which may or may not show up in the armory depending on how the spawners are feeling).

**Comment by Imagebard** [comment]
@icantthinkofanameritenow 
Okay but consider this:
Set the turret's emagged var to 1

**Comment by icantthinkofanameritenow** [comment]
@Imagebard 
Did that. 

Also, good news, bad news moment. Bad news is that I realized pulse shots won't work because they possess destroy and would shoot through the rwalls. So that's out.

The good news is that I found another type of beam called a "death laser" in the code that not only lacks the DESTROY flag but also does even more damage than the pulse shots.

Yeah, you're really going to want that ablative armor, or at least carry a shitload of dermaline.

**Comment by Imagebard** [comment]
Laser Cannons are 40, Pulse Shots are 50-60. *How much is a death laser?*

**Comment by Boxe** [comment]
death laser sounds like it might be the mecha pulse rifle one but iunno

**Comment by icantthinkofanameritenow** [comment]
@Imagebard 
According to https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/modules/projectiles/projectile/beams.dm

it's exactly 60, right on par with the Marauder's pulse rifle. The latter uses a different projectile type though for some reason. Weird.

**Comment by coldcola** [comment]
this is pretty large now, are the odds this spawns very rare?
it's size may make it pretty easy to find, as a thought.

**Comment by icantthinkofanameritenow** [comment]
@coldcola
I would prefer that it be made rarer than the other vault types, but I do not know how to go about doing that.

**Comment by ihadtoregisterforthis** [comment]
Don't go ezpz on the security, there's a lot of loot.

**Comment by icantthinkofanameritenow** [comment]
@ihadtoregisterforthis
You think I have room for a couple more Syndicate mobs in here?

**Comment by ihadtoregisterforthis** [comment]
Give it a trial run, even as invincible, and you'll know if it needs more mobs or not.

**Comment by icantthinkofanameritenow** [comment]
I'll be sure to do that. But refresh my memory, how will I be able to do so? I know that I need to go into Dream Daemon and activate the private server, but that's all I can remember off the top of my head.

File: vgstation13.dme [hunk 1]
```diff
@@ -1932,6 +1932,7 @@
 #include "maps\defficiency\areas.dm"
 #include "maps\defficiency\pipes.dm"
 #include "maps\randomvaults\objects.dm"
+#include "maps\randomvaults\oldarmory.dmm"
```
Comment (icantthinkofanameritenow) [inline vgstation13.dme]: Is this supposed to be here?

Comment (unid15) [inline vgstation13.dme]: Nope, delete it

**Comment by ihadtoregisterforthis** [comment]
Set yourself as admin in the config file, compile, open with dream daemon, and enter with the yellow sign button.

File: code/modules/randomMaps/vaults.dm [hunk 2]
```diff
@@ -14,12 +14,13 @@ var/const/vault_map_directory = "maps/randomVaults/"
...
 var/list/vault_map_names = list( //Add your vaults' map names to this list. Don't include the .dmm prefix
-	"icetruck_crash",
-	"asteroid_temple",
+	//"icetruck_crash",
+	//"asteroid_temple",
 	//"doomed_satelite", WARNING: this map is possessed. Uncommenting will cause bugs and general spookiness. Don't uncomment
-	"hivebot_factory",
-	"clown_base",
-	"rust"
+	//"hivebot_factory",
```
Comment (icantthinkofanameritenow) [inline code/modules/randomMaps/vaults.dm]: Commenting these out is for testing purposes only to ensure I can get the vault to actually spawn.

**Comment by icantthinkofanameritenow** [comment]
Testing completed. The turrets had to be replaced with regular emagged turrets because they didn't seem to take the modifications I made into account, but given how quickly all the commandos were robusting me even as a test dummy, I can say that just getting in there will be a tricky feat in itself. Besides, if the beacon explodes on you it'll probably kill you anyway.

**Comment by Imagebard** [comment]
Merge when?

**Comment by Boxe** [comment]
for some reason this gives me a very bad feeling

File: maps/randomvaults/oldarmory.dmm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+"ce" = (/obj/structure/sign/securearea,/turf/simulated/wall/r_wall,/area/vault/oldarmory)
+"cf" = (/obj/structure/painting/random,/turf/simulated/wall/r_wall,/area/vault/oldarmory)
+"cg" = (/obj/machinery/door/airlock/command{name = "Computer Chamber"},/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
+"ch" = (/obj/machinery/vending/wallmed1,/turf/simulated/wall/r_wall,/area/vault/oldarmory)
+"ci" = (/obj/machinery/vending/suitdispenser,/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
+"cj" = (/obj/structure/safe,/obj/item/toy/gasha/newcop,/obj/item/toy/syndicateballoon,/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
+"ck" = (/obj/machinery/door/airlock/vault{locked = 1; name = "Communications Relay"},/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
+"cl" = (/obj/machinery/vending/shoedispenser,/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
+"cm" = (/obj/machinery/syndicate_beacon{desc = "This seems to be what the Syndicate agents here are using to communicate with their leaders. Maybe they'd be willing to hire you?"; name = "Syndicate Communications Relay"},/turf/simulated/shuttle/floor4,/area/vault/oldarmory)
```
Comment (ghost) [inline maps/randomvaults/oldarmory.dmm:117]: What does the syndicate beacon even do?

Comment (icantthinkofanameritenow) [inline maps/randomvaults/oldarmory.dmm:117]: It's a very rare Xenoarchaeology find. If you're not an antagonist, it's a 50% chance of becoming a traitor with two objectives, with the first one being something more unusual than the normal objectives like "cut power to 80% or more of the station". (The other half is the beacon blowing up in your face.) Yes, it provides you with an uplink as well. 

If you're already an antagonist, it doesn't do shit.

**Comment by clusterfack** [comment]
Convert to unid's new system

**Comment by icantthinkofanameritenow** [comment]
Done. Now hurry the fuck up and merge this thing, I'm sick of all these conflicts occurring because you haven't merged it fast enough.

**Comment by clusterfack** [comment]
@icantthinkofanameritenow Cry more ya fucking baby

**Comment by icantthinkofanameritenow** [comment]
@clusterfack 
Just saying, all the conflicts I've run into thus far have been the direct result of it being merged too slowly.

**Comment by clusterfack** [comment]
@icantthinkofanameritenow No its a direct result of people making PRs touching the same code. The speed at which we merge things has nothing to fucking do with it.

**Comment by ihadtoregisterforthis** [comment]
Faster merges means less people touch that same code, and less conflicts caused by that touching

**Comment by clusterfack** [comment]
@ihadtoregisterforthis What do you think the job of the codebase is? To make the least work for coders and get out the fastest merges possible? I highly doubt that is your position.

**Comment by ihadtoregisterforthis** [comment]
@clusterfack The speed of the merge does affect the conflicts when the code changed is under many revisions. If the PR has no problems after being reviewed, prolonging the merge will make the author salty over time. That is my position.

**Comment by clusterfack** [comment]
@ihadtoregisterforthis As I said, we could immediately merge any new content immediately, but we don't for fucking OBVIOUS REASONS.

**Comment by ihadtoregisterforthis** [comment]
Not all the time at least ;^)

**Comment by clusterfack** [comment]
Whats the last new content that you think qualifies as that, honestly I'd like to know

**Comment by ihadtoregisterforthis** [comment]
Specifically tagged as "content", the newspapers catching fire, for example. Merged in nine minutes.
Another not so extreme could be the super singulo reborn, merged the next day and it's an important change without changing balance.

What's important here is, does this need anything else before being merged?

**Comment by clusterfack** [comment]
Newspaper was fixing a bug and also merged by someone other than me. 

Also you seem splitting hairs if one day is too short and four days is too long. These aren't significant differences of time.

**Comment by clusterfack** [comment]
But anyway the reason I left this up is because it adds a ton of loot that's normally unobtainable to the game as obtainable items. So I wanted it to have a large amount of time to gather player feedback both good or bad.

**Comment by Imagebard** [comment]
Nobody seems to have a problem with it so far. Well, unless you have something to say about it.
In which case, everyone's waiting.

**Comment by icantthinkofanameritenow** [comment]
@clusterfack 
Keep in mind that the loot has a considerable degree of randomness to it. There's some possibility that only a small handful of unobtainable items will actually appear in any given occurrence of the vault.

**Comment by icantthinkofanameritenow** [comment]
Obviously nobody's objecting to it at this point. So merge when? 

Speak now or forever hold your tongue.

**Comment by gbasood** [comment]
> Obviously nobody's objecting to it at this point. 

I still think it's a bit of an eyesore but that's just me.
Everyone else seems to like it. I'll stare at this tomorrow and consider a merge if someone doesn't get to it before me.

**Pull request merged**